scalableminds/webknossos
# Add OIDC authentication

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Create a local OIDC provider. See the comment below. (note: if keycloak is started via docker (not compose), change application.conf to `providerURL = "http://localhost:8080/realms/master/"`)
- also set `oidcEnabled` to true in the application.conf `features` block
- open http://localhost:9000/auth/login and use the "login with sso" button
- you should be redirected to the keycloak UI
- log in with admin:admin
- you should be redirected to webknossos (a new account was created and you are logged in)
- log out and log back in (e.g. with incognito tab), the same account should be logged in
- disable oidcEnabled again and verify that the log in/out mechanism works as usual

### Open questions:
- Is client authentication required (currently not implemented)
- [X] Should there be an option to disable OIDC?

### Issues:
- fixes #6437 

------
(Please delete unneeded items, merge only when none are left open)
